Source/WebCore/page/OpportunisticTaskScheduler.cpp

```diff
@@ -98,14 +117,16 @@ void OpportunisticTaskScheduler::runLoopObserverFired()
     };
 
     auto deadline = std::exchange(m_currentDeadline, MonotonicTime { });
-    page->opportunisticallyRunIdleCallbacks();
-    if (UNLIKELY(!page))
-        return;
+    if (std::exchange(m_mayHavePendingIdleCallbacks, false)) {
+        checkedPage()->opportunisticallyRunIdleCallbacks();
+        if (UNLIKELY(!m_page))
+            return;
+    }
 
-    if (!page->settings().opportunisticSweepingAndGarbageCollectionEnabled())
+    if (!checkedPage()->settings().opportunisticSweepingAndGarbageCollectionEnabled())
```

Comment:
We may want to store the page in a local variable to avoid repeated calls to checkedPage().